packages/components/src/DataViewer/RecordsViewer/Leaf/RecordsViewerLeaf.component.js

```diff
@@ -11,20 +12,57 @@ export function RecordsViewerLeaf({
 	className,
 	nodeHighlighted,
 	displayTypes,
+	measure,
 }) {
+	const ref = React.createRef();
+	const [isValueOverflown, setIsValueOverflown] = useState(false);
+	const [isLongValueToggled, setIsLongValueToggled] = useState(false);
+
+	useLayoutEffect(() => {
+		if (ref.current.offsetParent.offsetWidth < ref.current.scrollWidth) {
+			setIsValueOverflown(true);
+		}
+	}, []);
+
+	useLayoutEffect(() => {
+		measure();
+	}, [isLongValueToggled]);
+
 	return (
 		<div
 			className={classNames(theme['tc-records-viewer-leaf'], 'tc-records-viewer-leaf', className, {
 				[theme['tc-records-viewer-leaf-highlighted']]: nodeHighlighted,
 				'tc-records-viewer-leaf-highlighted': nodeHighlighted,
 			})}
 		>
+			{isValueOverflown && (
+				<span
+					className={classNames(theme['tc-leaf-overflow-icon'], 'tc-leaf-overflow-icon', className)}
+				>
+					<Icon
```

Comment:
The button is not accessible because of the alternative text here too.
But the issue is on Icon itself first, it doesn't allow a text. You can use ActionButton with the
* `icon` + `iconTransform` props to set the icon
* `label` + `hideLabel` props to have a hidden accessible label
* `link` props to have only the icon style. You surely will have to fix the margins/paddings, but you're already do that